Add Kerberos dependency to hive provider

kesem0811 opened the pull request:
Add Kerberos (GSSAPI) dependency to Hive provider
When using Python 3.12, the apache-airflow-providers-apache-hive provider relies on pure-sasl for SASL authentication.
If a user attempts to connect to Hive using Kerberos (GSSAPI) without having the required Kerberos libraries installed, they encounter the following error:
Could not start SASL: None of the mechanisms listed meet all required properties

After investigating and debugging, it was determined that this error occurs because pure-sasl requires the kerberos (or winkerberos on Windows) package to support GSSAPI, and it is not installed by default.
This PR updates the Hive provider’s dependencies to include pure-sasl[GSSAPI], ensuring that the necessary Kerberos support is installed automatically. This resolves the SASL startup error and allows users to connect to Hive using Kerberos without additional manual steps

kesem0811 committed:
Add Kerberos dependency to hive provider

- Add Kerberos>=1.3.0 as a required dependency
- Ensures kerberos package is installed when hive provider is installed
- Supports GSSAPI authentication for SASL connections

eladkal commented on providers/apache/hive/pyproject.toml:
```diff
@@ -65,6 +65,7 @@ dependencies = [
     "pyhive[hive_pure_sasl]>=0.7.0",
     "thrift>=0.11.0",
     "jmespath>=0.7.0",
+    "pykerberos>=1.1.13",
```
Should it be a dependency or an optional dependency?
I don't think everyone are using kerberos?

kesem0811 committed:
fix: update kerberos dependency in hive provider

- Replace pykerberos>=1.1.13 with Kerberos>=1.3.0
- Addresses package import issues with kerberos authentication

kesem0811 committed:
Move Kerberos to optional dependencies

- Remove Kerberos from main dependencies
- Add kerberos optional dependency group
- Users can now install with: pip install apache-airflow-providers-apache-hive[kerberos]
- Kerberos package only installed when explicitly requested

kesem0811 committed:
Fix formatting in pyproject.toml

- Remove trailing comma and newline in dependencies list
- Clean up formatting for better readability

kesem0811 committed:
Fix TOML syntax error in pyproject.toml

- Add missing comma after last dependency in dependencies array
- Fixes TOML parse error at line 67
- Resolves CI build failure

kesem0811 committed:
feat: add platform-specific Kerberos dependencies for GSSAPI

- Add conditional Kerberos library selection based on platform
- Windows: use winkerberos>=0.7.0 for better Windows compatibility
- Non-Windows: use kerberos>=1.3.0 for standard Kerberos support
- Improves cross-platform compatibility for Hive GSSAPI authentication
